src/util/importJSON.ts

```diff
@@ -0,0 +1,209 @@
+import _ from 'lodash'
+import { Child, Context, Lexeme, ParentEntry, Path } from '../types'
+import { State } from './initialState'
+import { EM_TOKEN, ROOT_TOKEN } from '../constants'
+import { getRankAfter, getThought, getThoughts, nextSibling } from '../selectors'
+
+// util
+import {
+  addThought,
+  contextOf,
+  createId,
+  equalPath,
+  equalThoughtRanked,
+  hashContext,
+  hashThought,
+  head,
+  headRank,
+  pathToContext,
+  removeContext,
+  rootedContextOf,
+  timestamp,
+} from '../util'
+import { Block } from '../action-creators/importText'
+import { GenericObject } from '../utilTypes'
+
+interface ImportHtmlOptions {
+  skipRoot? : boolean,
+}
+
+interface RankInfo {
+  rank: number,
+  deepLevel: number,
+}
+
+/** If skipRoot is true, replace head block with its children, or drop it, if head has no children. */
+const skipRootThought = (skipRoot: boolean, blocks: Block[]) => {
+  if (!skipRoot) return blocks
```

Comment:
A function should generally not include a dedicated boolean to bail. That logic should be located at the caller.